File: python/p3.py
# Solution for Project Euler problem 3
# http://projecteuler.net/index.php?section=problems&id=3

i = 2
f = 600851475143
roots = []
while f > 1:
    if f % i == 0:
        f /= i
        roots.append(i)
        continue
    i += 1
print roots

File: scala/p3.scala
// Solution for Project Euler problem 3
// http://projecteuler.net/index.php?section=problems&id=3

def factor(num: Long, fac: Int): Int = num match {
  case 1 => fac
  case x => if (x % fac == 0) factor(x/fac, fac) else factor(x, fac+1)
}

println(factor(600851475143L, 2))
